zkvm/src/types.rs

```diff
@@ -310,6 +323,25 @@ impl Expression {
     }
 }
 
+impl Neg for Expression {
+    type Output = Expression;
+
+    fn neg(self) -> Expression {
+        match self{
+            Expression::Constant(a) => {
+                Expression::Constant(-a)
+            }, 
+            Expression::Terms(terms,assignment)=> {
+
+                for (_, n) in terms.iter_mut() {
+                    -*n;
+                }
+                Expression::Terms(terms, assignment)
```

Comment:
assignment is not negated

Reply:
Would we need to add that in the `neg` section? https://github.com/interstellar/slingshot/blob/main/zkvm/spec/ZkVM.md#neg

Reply:
Spec is focused on verification only. The witness types are implementation detail. See also 
https://github.com/interstellar/slingshot/issues/96